include/rocksdb/file_system.h

```diff
@@ -749,13 +749,29 @@ class FSWritableFile {
   //
   // PositionedAppend() requires aligned buffer to be passed in. The alignment
   // required is queried via GetRequiredBufferAlignment()
+
+  // Append data with verification information
+  virtual IOStatus AppendWithVerify(
+      const Slice& data, const IOOptions& options, IODebugContext* dbg,
+      const DataVerificationInfo& /* verification_info */) {
+    return Append(data, options, dbg);
+  }
+
   virtual IOStatus PositionedAppend(const Slice& /* data */,
                                     uint64_t /* offset */,
                                     const IOOptions& /*options*/,
                                     IODebugContext* /*dbg*/) {
     return IOStatus::NotSupported();
   }
 
+  // PositionedAppend data with verification information.
+  virtual IOStatus PositionedAppendWithVerify(
+      const Slice& /* data */, uint64_t /* offset */,
+      const IOOptions& /*options*/, IODebugContext* /*dbg*/,
+      const DataVerificationInfo& /* verification_info */) {
```

Comment:
If `const DataVerificationInfo&` is passed in, then DataVerificationInfo is probably reused for multiple AppendWithVerify() calls. The probably less likely that DataVerificationInfo::checksum is set through constructor. Then think about whether it is needed to have a constructor with an input.